features/storage/FEATURE_STORAGE/TESTS/cfstore/create/create.cpp

```diff
@@ -828,7 +825,9 @@ Case cases[] = {
         Case("CREATE_test_03_end", cfstore_create_test_03_end),
         Case("CREATE_test_04_start", cfstore_utest_default_start),
         Case("CREATE_test_04_end", cfstore_create_test_04_end),
+#if defined(MBED_HEAP_STATS_ENABLED) && MBED_HEAP_STATS_ENABLED && !defined(__ICCARM__)
```

Comment:
The previous memory leak test was failing incorrectly because it couldn't tell the difference between slight fragmentation and a memory leak. This patch updates the code to reliably test on at least GCC_ARM and ARM using heap stats. Since IAR doesn't have heap stats this configuration isn't tested. It is also not tested if heap stats aren't enabled. If this is a problem the test should probably be removed.

One possible alternative it to estimate memory usage by calling malloc until out of space. This wouldn't be exact though, and might be a little on the hacky side.